# Crown_Frogs

## Directories  
A general guide to the directories and files contained in this repository.  

___

### Alignments
+ *Crown_Oz_Alignments.zip*: individual locus alignments e.g.
+ *Crown_Oz_CodonAlignments.zip*: codon position alignments, e.g. input alignment for MCMCtree is *CrownFrogs_Concat_NT_12_3.phy*
+ concatenated alignment *CrownFrogs_Concat.phy*
+ partition files for locus *CrownFrogs_Concat_partitions.txt* and codon position/NT *CrownFrogs_NTpartitions.txt*
+ locus summary file generated by AMAS *CrownFrogs_Oz_summary.txt*
+ *Intermediate_Alignments.zip* and *MMHH_Alignments.zip* contain preliminary alignment files from the different AHE projects prior to combining them
+ *RBH_AHE_XenopusExons.csv*: results of the Reciprocal Blast Hits of AHE against Xenopus exons
+ *Xenopus_tropicalis_exons.txt.zip*: zipped fasta file of Xenopus exons

___

### Biogeography
+ *BGB_DEC_...pdf*: Plots of 12 models (6 models x 2 hypotheses) from BioGeoBEARS
+ *BGB_DEC.R*: Code for running BioGeoBEARS
+ *BGB_dispersal.csv*: time-stratified dispersal multipliers among regions (for parameter *w*). Input file for BGB is *CrownFrogs_dispersal.txt*, not this file.
+ *BGB_distances_scaled.csv*: time-stratified distance multipliers among regions. Distances between physically adjacent regions were coded as 1, and then remaining distances were corrected by the shortest distance between two areas (for parameter *x*). Input file for BGB is *CrownFrogs_distances_scaled.txt*, not this file.
+ *BGB_Legend.pdf*: Color legend of areas and combinations
+ *BGB_Results_Fossils*: model comparison results
+ *CrownFrogs_BGB_Fossils_rotated.tre*: tree rotated for presentation in Fig.3
+ *CrownFrogs_BGB_Fossils.tre*: tree used for BGB analyses
+ *CrownFrogs_BGB_H1_ForPlotting.csv*: sample distributions for plotting
+ *CrownFrogs_BGB_H1.txt / CrownFrogs_BGB_H2.txt*: sample ranges for BGB analyses under H1 & H2
+ *CrownFrogs_DEC...Rdata*: BGB model results for 12 models (6 models x 2 hypotheses)
+ *CrownFrogs_dispersal.txt*: time-stratified dispersal multipliers among regions (for parameter *w*). File for use in BGB analysis
+ *CrownFrogs_distances_scaled.txt*: time-stratified distance multipliers among regions. Distances between physically adjacent regions were coded as 1, and then remaining distances were corrected by the shortest distance between two areas (for parameter *x*). File for use in BGB analysis.
+ *CrownFrogs_RangesKey.txt*
+ *CrownFrogs_timeperiods.txt*: time-stratifications for use in BGB analysis
+ *Fossil_Tree.R*: code for building tree incorporating fossil information for BGB analysis
+ *Plot_Observed_Ranges.R*: code to plot the contemporary distributions of samples (see Fig.3)

___

### Figures
+ Main text figures 1, 2, & 3 (*Fig.1*, ...)
+ Supplementary figures S1-7 (*Fig.S1*, ...)
+ Frog illustrations from main figure 2 (feel free to reuse!)
+ Fossil calibration figure from Table S2
+ *Globe_100-13Mya.ai*: Adobe Illustrator file of globe at relevant timeslices (Fig.3), exported from GPlates

___

### Heterotachy
+ *GTR_H...zip*: Results of varied number of H classes under the GHOST model
+ *CrownFrogs_Concat.phy.zip*: zipped phylip of concatenated alignment
+ *GHOST_ModelClasses.csv*: summary of GHOST model likelihoods

___

### Manuscript
+ Cover letter, manuscript, and presubmission notes
+ Response to reviewers, revised manuscript with (MARKUP) and without (CLEAN) changes

___

### Scripts
+ *AHE_Xenopus_ReciprocalBlast.R*: reciprocal blast script for identifying exonic AHE loci
+ *Alignment_Prep.R* is a scratchpad of the commands for combining the alignments from the four different AHE projects
+ *ASH_Anuran_SpeciesList.csv*: Australian Society of Herpetologists list of reconized Australian frog species
+ *combineOrthologues.R* function for combining alignments
+ *Comparative_Radiations.R* is the script for building the figure comparing Australian radiation ages by species richness
+ *Condensing_Alignments.R* helps to generate a pseudo-reference genome for each AHE project because the AHE locus numbers differ
+ *dropEmptyTaxa.R* removes taxa with all 'N', '-', or '?' in alignments. This happens when you split up a concatenated alignment
+ *ggplotRegression.R* linear regression plotted as ggplot object
+ *Oz_Rainfall_Map.R* is the script for building the rainfall map of Australia
+ *plot_BGB.R*: adapted BioGeoBEARS script for plotting BGB output
+ *Richness_Maps.R*: code for generating hex maps of species richness as in Fig.1

___

### Trees
+ *Crown_Oz_AllLoci.trees*: genetrees for all loci
+ *Crown_Oz_ASTRAL_Names.csv*: file of sample names
+ *Crown_Oz_ASTRAL.tre*: ASTRAL species tree built from IQTREE genetrees
+ *Crown_Oz_GeneTrees.zip*: named genetree files
+ *CrownFrogs_Calibrations.tre*: input tree for MCMCtree labelled with fossil calibrations at nodes
+ *CrownFrogs_Concat_GTRH4.contree*: output of GHOST GTR+H4 concatenated species tree
+ *CrownFrogs_mcmctree.tre*: MCMCtree time-calibrated tree (Fig.2)
+ *CrownFrogs_Oz_qDist20_ASTRAL.tre*: ASTRAL species tree build from IQTREE genetrees with quartet dissimilarity scores of <20 from *Crown_Oz_Astral.tre*
+ *CrownFrogs_Oz_qDist20.trees*: IQTREE genetrees of loci with quartet dissimilarity scores <20 from *Crown_Oz_Astral.tre*. 


