# Scripts
+ *AHE_Xenopus_ReciprocalBlast.R*: reciprocal blast script for identifying exonic AHE loci
+ *Alignment_Prep.R* is a scratchpad of the commands for combining the alignments from the four different AHE projects
+ *ASH_Anuran_SpeciesList.csv*: Australian Society of Herpetologists list of reconized Australian frog species
+ *combineOrthologues.R* function for combining alignments
+ *Comparative_Radiations.R* is the script for building the figure comparing Australian radiation ages by species richness
+ *Condensing_Alignments.R* helps to generate a pseudo-reference genome for each AHE project because the AHE locus numbers differ
+ *dropEmptyTaxa.R* removes taxa with all 'N', '-', or '?' in alignments. This happens when you split up a concatenated alignment
+ *ggplotRegression.R* linear regression plotted as ggplot object
+ *Oz_Rainfall_Map.R* is the script for building the rainfall map of Australia
+ *plot_BGB.R*: adapted BioGeoBEARS script for plotting BGB output
+ *Richness_Maps.R*: code for generating hex maps of species richness as in Fig.1

___

